tools/automator.sh

```diff
@@ -96,7 +96,7 @@ function validate_args() {
   TOKEN="$(cat "${TOKEN_FILE}")"
   [ -z "${TOKEN}" ] && usage "Invalid token file"
 
-  BRANCH="${BRANCH:-$(git describe --contains --all HEAD)}"
+  BRANCH="${BRANCH:-$(git symbolic-ref --quiet --short HEAD)}"
```

Comment:
hmm.. when I run that on a repo that is in detached-head state I get empty string:

```bash
16:10:38 t580-rh {MAISTRA-1898} ~/dev/go/src/istio.io/istio$ git checkout 693f2fdcce7fcdc11925af2f0dd07259a57e899e
Note: switching to '693f2fdcce7fcdc11925af2f0dd07259a57e899e'.

You are in 'detached HEAD' state. [...]

HEAD is now at 693f2fdcce Fix merge logic and conflicts in push_context.go
16:10:41 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$ git describe --contains --all HEAD
MAISTRA-1898~1
16:10:43 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$ git symbolic-ref --quiet --short HEAD
16:10:46 t580-rh {(693f2fdcce...)} ~/dev/go/src/istio.io/istio$
```

Reply:
yes. same here. This is not supposed to run in a detached head

Reply:
btw, upstream ran into the same issue... https://github.com/istio/test-infra/pull/2611

Reply:
ah I see, thanks